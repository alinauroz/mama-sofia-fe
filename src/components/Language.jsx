import React from 'react'

const lang = localStorage.getItem("lang") || "en"

export default class Language extends React.Component {
    constructor (props) {
        super(props);
        this.state = {
            langs: ["en", "fr"]
        }
    }

    onChange_ = (e) => {
        localStorage.setItem("lang", e.target.value);
        window.location.reload();
    }

    render () {
        return (
            <select 
                style={{marginTop: 10}}
                onChange = {this.onChange_}
                value = {lang}
            >
                {
                    this.state.langs.map(l => {
                        return <option key={l}> {l} </option>
                    })
                }
            </select>
        )
    }
}