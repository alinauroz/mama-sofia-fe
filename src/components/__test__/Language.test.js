import React from 'react';
import ReactDOM from 'react-dom'
import Language from './../Language.jsx'
import renderer from 'react-test-renderer'

it ('renders without crashing', () => {
    const div = document.createElement("div");
    ReactDOM.render(<Language />, div);
});

it ('matches snapshot', () => {
    const tree = renderer.create(
        <Language/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
})