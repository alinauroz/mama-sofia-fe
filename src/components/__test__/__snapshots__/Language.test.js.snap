// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<select
  onChange={[Function]}
  style={
    Object {
      "marginTop": 10,
    }
  }
  value="en"
>
  <option>
     
    en
     
  </option>
  <option>
     
    fr
     
  </option>
</select>
`;
